### PR TITLE
FIX: Exact same file after switching submissions

### PR DESCRIPTION
**Pull request by DecathectZero**

### CODE REVIEW

File: app/views/results/common/_file_selector_dropdown.html.erb [hunk 1]
```diff
@@ -14,23 +14,33 @@
...
-            } else {
-              first_file_id = <%= files[0].id %>;
-              first_file_path = '<%= files[0].path %>/<%= files[0].filename %>';
+        <% if @current_user.ta? && (assignmentfile =
+              AssignmentFile.select(:filename)
+              .joins('JOIN criteria_assignment_files_joins c ON c.assignment_file_id = assignment_files.id')
+              .joins('JOIN criterion_ta_associations t ON c.criterion_id = t.criterion_id')
+              .where('t.ta_id = ? AND assignment_files.assignment_id = ?', @current_user.id, @assignment.id)
+              .first) %>
+        <% index = files.find_index{|x| x.filename == assignmentfile.filename} %>
```
Comment (david-yz-liu) [inline app/views/results/common/_file_selector_dropdown.html.erb]: Please indent the lines of code under the `if` and `else`.

File: app/views/results/common/_file_selector_dropdown.html.erb [hunk 2]
```diff
@@ -14,23 +14,33 @@
...
+        first_file_id = <%= files[index].id %>;
+        first_file_path = '<%= files[index].path %>/<%= files[index].filename %>';
+        <% else %>
+        $(document).ready(function(){
```
Comment (david-yz-liu) [inline app/views/results/common/_file_selector_dropdown.html.erb]: Missing some whitespace here

File: app/views/results/common/_file_selector_dropdown.html.erb [hunk 3]
```diff
@@ -14,23 +14,33 @@
...
+          first_file_id = <%= files[0].id %>;
+          first_file_path = '<%= files[0].path %>/<%= files[0].filename %>';
+          if (localStorage.getItem('assignment_id') == <%= @assignment.id %> && localStorage.getItem('file_path')) {
```
Comment (david-yz-liu) [inline app/views/results/common/_file_selector_dropdown.html.erb]: In general use `===` not `==` (this occurs below as well).

Comment (DecathectZero) [inline app/views/results/common/_file_selector_dropdown.html.erb]: Some of these variables have different types, is it still better to use === and convert the types before comparison?

Comment (david-yz-liu) [inline app/views/results/common/_file_selector_dropdown.html.erb]: Great question. Yes, this is still strongly preferred. Note that you can kind of cheat in the Ruby code and do something like `'<%= @asignment.id %>'` to get a string.

**Pull request merged**